### ISSUE TITLE
hyphen and minus

### ISSUE THREAD

**Issue by unc0** [kkirby/gorillascript#1]
> 1. Remove the support for dashed identifiers as it can cause confusion: Example: a-b, should it mean variable a-b, or a minus b? We should use the latter.

I use lots of "dashed identifiers" in my gs projects. It came from lisp, and you don't have to SHIFT camelCase.

```
a-var // for identifiers
a - var // for minus
```

**Comment by unc0** [comment]
And I think we should fix `require!` macro, lots of npm packages name uses hyphen.

```
require! foo-bar
```

will translate to

``` javascript
var fooBar = require('fooBar');
```

But it should be

``` javascript
var fooBar = require('foo-bar');
```

**Comment by kkirby** [comment]
Hm, then I think at the very least there should be an option to turn them off, as some people's coding style doesn't have white space between operators.

As for the require macro, that's actually probably a really simple fix. When I get some time I'll take a look.

**Comment by unc0** [comment]
Yes, this and the bit operators override should be optional. I still believe gorillascript should be shrunk down to a minimal form. For example, the compose operator (`|>`) isn't that useful if you are working with an OOP-style API.

**Comment by unc0** [comment]
```
## operators: +fp
// make syntax options looks like C define.
// enable functional programming operators
// this still valid GS syntax: operators: { fp: true }
```
